src/nools/task-emitter.js

```diff
@@ -120,10 +120,17 @@ function emitTasks(taskDefinition, Utils, Task, emit, c, r) {
         readyStart: event.start || 0,
         readyEnd: event.end || 0,
         title: taskDefinition.title,
-        resolved: taskDefinition.resolvedIf(c, r, event, dueDate, scheduledTaskIdx),
```

Comment:
If we do the task defaults file (from above comment) then that means taskDefinition already has the `defaultResolvedIf()` then you can use it from there: `taskDefinition.defaultResolvedIf(....)`

We can make a function that will handle that check. Just to give you an example:
```
callResolvedIf(.......) {
   if (.....) {
       return taskDefinition.resolvedIf(...);
   }

  return taskDefinition.defaultResolvedIf(...);
}

// then in the task def
task = {
   ... other props ...

   resolved: callResolvedIf(....),  
};
```

Because `emitForEvents` function is growing a lot and doing too many things, it should just emit and not do validations or other logic, but we'll fix it little by little as we go (not in this PR), so that's why I was thinking of reducing the default fn logic in this place and take it somewhere out. 
What do you think?

Reply:
I think this will work great. Thanks.